src/kvstore/Common.h

```diff
@@ -39,6 +40,183 @@ class KVFilter {
                       const folly::StringPiece& val) const = 0;
 };
 
+/**
+ * @brief Part peer, including normal peer and learner peer.
+ *
+ */
+struct Peer {
+  enum class Status : uint8_t {
+    kNormalPeer = 0,
+    kLearner = 1,
+    kPromotedPeer = 2,
+    kDeleted = 3,
+    kMax = 4,
+  };
+
+  HostAddr addr;
+  Status status;
+
+  Peer() : addr(), status(Status::kNormalPeer) {}
+  Peer(HostAddr a, Status s) : addr(a), status(s) {}
+  Peer(const Peer& other) : addr(other.addr), status(other.status) {}
+
+  std::string toString() const {
+    return addr.toString() + "," + std::to_string(static_cast<int>(status));
+  }
+
+  void fromString(const std::string& str) {
+    auto pos = str.find(",");
+    if (pos == std::string::npos) {
+      LOG(ERROR) << "Parse peer failed:" << str;
+      return;
+    }
+    addr = HostAddr::fromString(str.substr(0, pos));
+
+    int s = std::stoi(str.substr(pos + 1));
+    if (s >= static_cast<int>(Status::kMax) || s < 0) {
+      LOG(ERROR) << "Parse peer status failed:" << str;
+      return;
+    }
+
+    status = static_cast<Status>(s);
+  }
+
+  bool operator==(const Peer& rhs) const {
+    return addr == rhs.addr && status == rhs.status;
+  }
+
+  bool operator!=(const Peer& rhs) const {
+    return !(*this == rhs);
+  }
+
+  Peer& operator=(const Peer& rhs) {
+    addr = rhs.addr;
+    status = rhs.status;
+    return *this;
+  }
+
+  static Peer nullPeer() {
+    return Peer();
+  }
+};
+
+inline std::ostream& operator<<(std::ostream& os, const Peer& peer) {
+  return os << peer.toString();
+}
+
+/**
+ * @brief Peers for one partition, it should handle serializing and deserializing.
+ *
+ */
+struct Peers {
+  std::map<HostAddr, Peer> peers;
```

Comment:
You could make it private, since you offer the get/remove/... interfaces.

Reply:
Make it private or remove the interface, which would be better?

Reply:
It is a personal preference, I prefer make the data member private.

Reply:
OK